src/provisioning/api.md

```diff
@@ -0,0 +1,290 @@
+    "config": {
+        // ... connection specific details, can be configured.
+    }
+}]
+```
+
+
+## GET /v1/{roomId}/connections/{id}
+
+Request details of a single connection. The `{roomId}` parameter is the target Matrix room.
+
+### Response
+
+```json5
+{
+
+    "type": "JiraProject", // The name of the connection
+    "eventType": "uk.half-shot.matrix-hookshot.jira.project", // Corresponds to the state type in the connection
+    "id": "opaque-unique-id", // An opaque ID used to refer to this connection. Should **NOT** be assumed to be stable.
+    "service": "jira", // or github, webhook. A human-readable service name to make things look pretty
+    "botUserId": "@hookshot:yourdomain.com", // The bot mxid for the service. Currently this is the sender_localpart, but may change in the future.
+    "config": {
+        // ... connection specific details, can be configured.
+    }
+}
+```
+
+## PUT /v1/{roomId}/connections/{type}
+
+Create a new connection of a given type. The type refers to the `IConnection.CanonicalEventType`. The `{roomId}` parameter is the target Matrix room.
+
+The body of the request is the configuration for the connection, which will be the "ConnectionState" interface for each connection.
+
+The request will respond with a `202` on success, as the connection creation process is asyncronous (being driven by Matrix state).
+
+### Request body
+```json5
+{
+    // ... connection specific details, can be configured.
+}
+```
+### Response
+
+```json5
+{
+    // The eventId of the state event that describes the connection.
+    "eventId": "!abc:def"
+}
+```
+
+## PATCH /v1/{roomId}/connections/{id}
+
+Update a connection's configuration. The `id` refers to the `id` returned in the GET response.
+
+The body of the request is the configuration for the connection, which will be the "ConnectionState" interface for each connection.
+
+### Request body
+```json5
+{
+    // ... connection specific details, can be configured.
+}
+```
+### Response
+
+```json5
+{
+
+    "type": "JiraProject", // The name of the connection
+    "eventType": "uk.half-shot.matrix-hookshot.jira.project", // Corresponds to the state type in the connection
+    "id": "opaque-unique-id", // An opaque ID used to refer to this connection. Should **NOT** be assumed to be stable.
+    "service": "jira", // or github, webhook. A human-readable service name to make things look pretty
+    "botUserId": "@hookshot:yourdomain.com", // The bot mxid for the service. Currently this is the sender_localpart, but may change in the future.
+    "config": {
+        // ... connection specific details, can be configured.
+    }
+}
+```
+
+## DELETE /v1/{roomId}/connections/{id}
+
+Delete a connection. The `id` refers to the `id` returned in the GET response.
+### Response
+
+```json5
+{
+    "ok": true
+}
+```
+
+# Service specific APIs
+
+Some services have specific APIs for additional functionality, like OAuth.
+
+## GitHub
+
+
+### GET /v1/github/oauth?userId={userId}
```

Comment:
The URL returned by this works great for authorizing the user, but I think if the github app hasn't been added to the org (including self-org like `turt2live`) then the bridge won't be able to list orgs.

Can we get an endpoint for the second step please? 😇